tests/models/test_implicit_bpr.py

```diff
@@ -0,0 +1,435 @@
+                    }
+                ),
+            ),
+        ),
+    )
+    def test_basic(
+        self,
+        dataset: Dataset,
+        filter_viewed: bool,
+        expected_cpu: pd.DataFrame,
+        expected_gpu: pd.DataFrame,
+        use_gpu: bool,
+    ) -> None:
+        base_model = BayesianPersonalizedRanking(
+            factors=2, num_threads=2, iterations=100, use_gpu=use_gpu, random_state=42
+        )
+        self._init_model_factors_inplace(base_model, dataset)
+        model = ImplicitBPRWrapperModel(model=base_model).fit(dataset)
+        actual = model.recommend(
+            users=np.array([10, 20]),
+            dataset=dataset,
+            k=2,
+            filter_viewed=filter_viewed,
+        )
+        expected = expected_gpu if use_gpu else expected_cpu
+        pd.testing.assert_frame_equal(actual.drop(columns=Columns.Score), expected)
+        pd.testing.assert_frame_equal(
+            actual.sort_values([Columns.User, Columns.Score], ascending=[True, False]).reset_index(drop=True),
+            actual,
+        )
+
+    def test_consistent_with_pure_implicit(self, dataset: Dataset, use_gpu: bool) -> None:
+        base_model = BayesianPersonalizedRanking(
+            factors=2, num_threads=2, iterations=100, use_gpu=use_gpu, random_state=42
+        )
+        self._init_model_factors_inplace(base_model, dataset)
+        users = np.array([10, 20, 30, 40])
+
+        model_for_wrap = deepcopy(base_model)
+        state = np.random.get_state()
+        wrapper_model = ImplicitBPRWrapperModel(model=model_for_wrap).fit(dataset)
+        actual_reco = wrapper_model.recommend(users=users, dataset=dataset, k=3, filter_viewed=False)
+
+        ui_csr = dataset.get_user_item_matrix(include_weights=True)
+        np.random.set_state(state)
+        base_model.fit(ui_csr)
+        for user_id in users:
+            internal_id = dataset.user_id_map.convert_to_internal([user_id])[0]
+            expected_ids, expected_scores = base_model.recommend(
+                userid=internal_id,
+                user_items=ui_csr[internal_id],
+                N=3,
+                filter_already_liked_items=False,
+            )
+            actual_ids = actual_reco.loc[actual_reco[Columns.User] == user_id, Columns.Item].values
+            actual_internal_ids = dataset.item_id_map.convert_to_internal(actual_ids)
+            actual_scores = actual_reco.loc[actual_reco[Columns.User] == user_id, Columns.Score].values
+            np.testing.assert_equal(actual_internal_ids, expected_ids)
+            np.testing.assert_allclose(actual_scores, expected_scores, atol=0.03)
+
+    @pytest.mark.parametrize(
+        "filter_viewed,expected",
+        (
+            (
+                True,
+                {10: {13, 17}, 20: {17}},
+            ),
+            (
+                False,
+                {10: {11, 13, 17}, 20: {11, 13, 17}},
+            ),
+        ),
+    )
+    def test_with_allowlist(
```

Comment:
Could you please rename this to `test_with_whitelist` to keep consistency within the framework?

Reply:
Okay.

I was trying to follow the [ACM guideline](https://www.acm.org/diversity-inclusion/words-matter#:~:text=friendly%20force%2C%20etc.-,blacklist/whitelist,-%3A%20terms%20used%20to). Using these terminologies could cause flagging on ethic concerns in the paper review such [as ICLR](https://openreview.net/forum?id=Z1MKx8mNu9&noteId=F5JZsdpYBN).

Reply:
Thank you for this comment. I agree, it's an important topic and we should using correct terminologies. I think we should create a separate PR, mark `whitelist` as deprecated everywhere, and add `allowlist`.

Reply:
Thanks for your suggestion. I committed to making it into a whitelist for now.
[69c92f4](https://github.com/MobileTeleSystems/RecTools/pull/232/commits/69c92f48990bd09eaaa70848dc3dd23629f1a2ab)